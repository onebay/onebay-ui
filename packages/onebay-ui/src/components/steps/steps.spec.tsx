import { mount } from '@vue/test-utils'
import Steps from './index'

const factory = (values = {}, slots = { default: [''] }) => {
  return mount(Steps, {
    slots,
    props: { ...values },
  })
}

const items1 = [
  {
    title: 'step1',
    desc: 'Here is additional information, up to two lines',
    icon: {
      value: 'sound',
      activeColor: '#fff',
      inactiveColor: '#78A4FA',
      size: '14',
    },
  },
  {
    title: 'step2',
    desc: 'Here is additional information, up to two lines',
    icon: {
      value: 'shopping-cart',
      activeColor: '#fff',
      inactiveColor: '#78A4FA',
      size: '14',
    },
  },
  {
    title: 'step3',
    desc: 'Here is additional information, up to two lines',
    icon: {
      value: 'camera',
      activeColor: '#fff',
      inactiveColor: '#78A4FA',
      size: '14',
    },
  },
]

describe('Steps Snap', () => {
  const items2 = [
    {
      title: 'step1',
      desc: 'Here is additional information, up to two lines',
      status: 'success'
    },
    {
      title: 'step2',
      desc: 'Here is additional information, up to two lines',
    },
    {
      title: 'step3',
      desc: 'Here is additional information, up to two lines',
      error: true,
    },
  ]

  it('render initial Steps', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Steps -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Steps -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Steps -- props items', () => {
    const wrapper1 = factory({ items: items1 })
    expect(wrapper1.element).toMatchSnapshot()
    const wrapper2 = factory({ items: items2 })
    expect(wrapper2.element).toMatchSnapshot()
  })

  it('render Steps -- props current', () => {
    const wrapper = factory({ items: items1, current: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })
})
describe('Steps Event', () => {
  it('Steps onChange', async () => {
    const onChange = jest.fn()
    const wrapper = factory({ items: items1, onChange })
    wrapper.find('.ob-steps__item').trigger('click')
    expect(onChange).toBeCalled()
  })
})
