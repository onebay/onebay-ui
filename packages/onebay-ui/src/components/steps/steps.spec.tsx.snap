// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps Snap render Steps -- props className 1`] = `
<div
  class="ob-steps test"
>
  
  
</div>
`;

exports[`Steps Snap render Steps -- props current 1`] = `
<div
  class="ob-steps"
>
  
  <div
    class="ob-steps__item ob-steps__item--inactive"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <!---->
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-icon ob-icon-sound ob-steps__circle-icon"
        />
      </div>
      <div
        class="ob-steps__right-line"
      />
    </div>
    <div
      class="ob-steps__title"
    >
      step1
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  <div
    class="ob-steps__item ob-steps__item--active"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <div
        class="ob-steps__left-line"
      />
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-icon ob-icon-shopping-cart ob-steps__circle-icon"
        />
      </div>
      <div
        class="ob-steps__right-line"
      />
    </div>
    <div
      class="ob-steps__title"
    >
      step2
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  <div
    class="ob-steps__item ob-steps__item--inactive"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <div
        class="ob-steps__left-line"
      />
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-icon ob-icon-camera ob-steps__circle-icon"
        />
      </div>
      <!---->
    </div>
    <div
      class="ob-steps__title"
    >
      step3
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  
</div>
`;

exports[`Steps Snap render Steps -- props style 1`] = `
<div
  class="ob-steps"
  style="color: red;"
>
  
  
</div>
`;

exports[`Steps Snap render Steps -- props items 1`] = `
<div
  class="ob-steps"
>
  
  <div
    class="ob-steps__item ob-steps__item--active"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <!---->
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-icon ob-icon-sound ob-steps__circle-icon"
        />
      </div>
      <div
        class="ob-steps__right-line"
      />
    </div>
    <div
      class="ob-steps__title"
    >
      step1
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  <div
    class="ob-steps__item ob-steps__item--inactive"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <div
        class="ob-steps__left-line"
      />
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-icon ob-icon-shopping-cart ob-steps__circle-icon"
        />
      </div>
      <div
        class="ob-steps__right-line"
      />
    </div>
    <div
      class="ob-steps__title"
    >
      step2
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  <div
    class="ob-steps__item ob-steps__item--inactive"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <div
        class="ob-steps__left-line"
      />
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-icon ob-icon-camera ob-steps__circle-icon"
        />
      </div>
      <!---->
    </div>
    <div
      class="ob-steps__title"
    >
      step3
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  
</div>
`;

exports[`Steps Snap render Steps -- props items 2`] = `
<div
  class="ob-steps"
>
  
  <div
    class="ob-steps__item ob-steps__item--active"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <!---->
      <div
        class="ob-icon ob-icon-check-circle ob-steps__single-icon ob-steps__single-icon--success"
      />
      <div
        class="ob-steps__right-line"
      />
    </div>
    <div
      class="ob-steps__title"
    >
      step1
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  <div
    class="ob-steps__item ob-steps__item--inactive"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <div
        class="ob-steps__left-line"
      />
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-steps__num"
        >
          2
        </div>
      </div>
      <div
        class="ob-steps__right-line"
      />
    </div>
    <div
      class="ob-steps__title"
    >
      step2
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  <div
    class="ob-steps__item ob-steps__item--inactive"
  >
    <div
      class="ob-steps__circular-wrap"
    >
      <div
        class="ob-steps__left-line"
      />
      <div
        class="ob-steps__circular"
      >
        <div
          class="ob-steps__num"
        >
          3
        </div>
      </div>
      <!---->
    </div>
    <div
      class="ob-steps__title"
    >
      step3
    </div>
    <div
      class="ob-steps__desc"
    >
      Here is additional information, up to two lines
    </div>
  </div>
  
</div>
`;

exports[`Steps Snap render initial Steps 1`] = `
<div
  class="ob-steps"
>
  
  
</div>
`;
